specification/sections/04-hooks.md

```diff
@@ -129,15 +132,17 @@ Evaluation context merge order is defined in [Context levels and merging](./03-e
 
 #### Requirement 4.3.6
 
-> The `after` stage **MUST** run after flag resolution occurs. It accepts a `hook context` (required), `flag evaluation details` (required) and `hook hints` (optional). It has no return value.
+> The `after` stage **MUST** run after flag resolution occurs. It accepts a `hook context` (required), `evaluation details` (required) and `hook hints` (optional). It has no return value.
 
 #### Requirement 4.3.7
 
 > The `error` hook **MUST** run when errors are encountered in the `before` stage, the `after` stage or during flag resolution. It accepts `hook context` (required), `exception` representing what went wrong (required), and `hook hints` (optional). It has no return value.
 
 #### Requirement 4.3.8
 
-> The `finally` hook **MUST** run after the `before`, `after`, and `error` stages. It accepts a `hook context` (required) and `hook hints` (optional). There is no return value.
+> The `finally` hook **MUST** run after the `before`, `after`, and `error` stages. It accepts a `hook context` (required), `evaluation details` (required) and `hook hints` (optional). It has no return value.
+
+The evaluation details passed to the `finally` stage should match the evaluation details returned to the application author.
```

Comment:
Maybe we can use a harder word than `should` here? Is there a reason to ever give back "non-matching" evaluation details?

Reply:
I generally try to avoid all RFC-2119 words in this sort of text, and instead use `present simple` tense:

`The evaluation details passed to the `finally` matches the evaluation details returned to the application author.